modules/social_features/social_tagging/social_tagging.module

```diff
@@ -277,14 +287,14 @@ function social_tagging_entity_delete(EntityInterface $entity) {
 /**
  * Build output on node view.
  *
- * @param \Drupal\node\NodeInterface $node
+ * @param \Drupal\Core\Entity\Entity $entity
  *   A node.
  *
- * @return array
+ * @return array|string
```

Comment:
Why do we return an empty string in some cases instead of just an empty array? This makes using the function more difficult because something like `foreach (social_tagging_process_tags($entity) as $tag)` will break without first checking which return type we have.

Reply:
Furthermore if this is done for display purposes then something like `join([], ", ")` would properly output an empty string already :)